server/views/formPages/incident/evidence.html

```diff
@@ -1,5 +1,337 @@
+                      })
+                    }} 
+    
+                  {% if loop.length != 1 %}
+                  <button type="button" class="govuk-button govuk-button--secondary add-another__remove-button">
+                    Remove
+                  </button>
+                  {% endif  %}
+                  
+                  {% endcall %} 
+                  {% else %} 
+                  {% call govukFieldset({ classes: 'add-another__item' }) %}
+                    
+                    {{
+                      govukInput({
+                        label: {
+                          html: 'Camera number'
+                        },
+                        classes: 'govuk-!-width-one-third',
+                        id: 'body-worn-camera-numbers[0][name]',
+                        name: 'bodyWornCameraNumbers[0][name]',
+                        attributes: {
+                          'data-name': 'bodyWornCameraNumbers[%index%][name]',
+                          'data-id': 'bodyWornCameraNumbers[%index%][name]'
+                        }
+                      })
+                    }}
+    
+                  {% endcall %} 
+                {% endfor %}
+                <div class="button-action">
+                  {{
+                    govukButton({
+                      text: 'Add another',
+                      classes: 'govuk-button--secondary  add-another__add-button govuk-!-margin-bottom-4',
+                      attributes: { 'data-qa-add-another-camera': true }
+                    })
+                  }}
+                </div> 
+              </div>  <!-- add another ends here -->
+            </div>  <!-- hidden panel end -->
+
+            <div class="govuk-radios__item">
+              <input class="govuk-radios__input" id="body-worn-camera-no" name="bodyWornCamera" type="radio"
+                value="no" 
+                {% if data.bodyWornCamera === 'no' %} checked {% endif %}
+              >
+              <label class="govuk-label govuk-radios__label" for="body-worn-camera-no">
+                No
+              </label>
+            </div>
+            <div class="govuk-radios__item">
+                <input class="govuk-radios__input" id="body-worn-camera-notKnown" name="bodyWornCamera" type="radio"
+                  value="notKnown" 
+                  {% if data.bodyWornCamera === 'no' %} checked {% endif %}
+                >
+                <label class="govuk-label govuk-radios__label" for="body-worn-camera-notKnown">
+                  Not known
+                </label>
+              </div>
+          </div>
+        </fieldset>     
+      </div> <!-- end of this Q4 component  -->
+
+   </div>   <!-- end of govuk-grid-column-full-->
+</div>    <!-- end of govuk-grid-row-->
+
+<script src="/assets/js/jquery.min.js"></script>
+<script src="/assets/add-another.js"></script>
+<script>
+new AddAnother($('.add-another-tag'))
+new AddAnother($('.add-another-camera'))
+
+</script>
+
+
+   
+
```

Comment:
Can you think of any way to break down this template into smaller, perhaps reusable, units? I am sure it does the job, but it seems quite long. Perhaps a per-question template?

Reply:
Might be worth thinking about ways to pull in the data values rather than embedding in the HTML?

Reply:
Yes, it is rather long. But I did it this way to stay consistent with other similar pages in the application. The other thing is, although the components do look the same, there are differences in the way they are laid out. For example, some are 2 radios horizontally, others are 3 vertically and then some expose hidden panels that display different numbers of input fields when clicking the 'yes' radio button.